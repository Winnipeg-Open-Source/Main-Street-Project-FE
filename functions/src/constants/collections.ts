export const DONATIONS_COLLECTIONS = 'Donations';
export const DONORS_COLLECTION = 'Donors';
export const EMPLOYEES_COLLECTION = 'Employees';
export const ITEMS_COLLECTION = 'Items';
export const LINE_ITEMS_COLLECTION = 'Line Items';
export const LOCATIONS_COLLECTION = 'Locations';
export const REQUISITIONS_COLLECTION = 'Requisitions';
