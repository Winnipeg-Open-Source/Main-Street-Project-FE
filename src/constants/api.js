export const BASE_API_PATH = 'http://localhost:8080';
export const DONORS_API_PATH = BASE_API_PATH + '/api/donors';
