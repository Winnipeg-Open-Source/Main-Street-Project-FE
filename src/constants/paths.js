export const LANDING_PATH = '/';
export const DONORS_PATH = '/donors';
export const DONOR_PATH = '/donors/:id';
export const PRODUCTS_PATH = '/products';
export const PRODUCT_PATH = '/products/:id';
export const PRODUCTS_CREATE_PATH = '/products/create';
export const PRODUCTS_EDIT_PATH = '/products/edit/:id';
export const DONATIONS_NEW_PATH = '/donations/new';
export const REQUISITION_NEW_PATH = '/requisition/new';

const getPath = (path, id) => {
    return id ? `${path}/${id}` : path;
};

export const getDonorPath = (id) => {
    return getPath(DONORS_PATH, id);
};

export const getProductPath = (id) => {
    return getPath(PRODUCTS_PATH, id);
};

export const getProductCreatePath = (id) => {
    return getPath(PRODUCTS_CREATE_PATH, id);
};

export const getProductEditPath = (id) => {
    return getPath(PRODUCTS_EDIT_PATH, id);
};