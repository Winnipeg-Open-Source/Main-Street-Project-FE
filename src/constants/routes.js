import React from 'react';
import { Document, ExitToApp, Gallery, Home, Loyalty, Map, User } from 'pcln-icons';
import {
    LANDING_PATH,
    DONATIONS_PATH,
    DONORS_PATH,
    ITEMS_PATH,
    LOCATIONS_PATH,
    LOGOUT_PATH,
    REQUISITIONS_PATH,
} from 'constants/paths';

const routes = [
    {
        label: 'Home',
        path: LANDING_PATH,
        icon: <Home />,
    },
    {
        label: 'Donors',
        path: DONORS_PATH,
        icon: <User />,
    },
    {
        label: 'Donations',
        path: DONATIONS_PATH,
        icon: <Loyalty />,
    },
    {
        label: 'Items',
        path: ITEMS_PATH,
        icon: <Gallery />,
    },
    {
        label: 'Locations',
        path: LOCATIONS_PATH,
        icon: <Map />,
    },
    {
        label: 'Requisitions',
        path: REQUISITIONS_PATH,
        icon: <Document />,
    },
    {
        label: 'Logout',
        path: LOGOUT_PATH,
        icon: <ExitToApp />,
    },
];

export default routes;
