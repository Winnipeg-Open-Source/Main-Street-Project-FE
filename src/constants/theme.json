{
    "palette": {
        "primary": {
            "light": "#FEFBF6",
            "base": "#E86754",
            "dark": "#9D2715"
        },
        "secondary": {
            "light": "#FAEBD7",
            "base": "#F0C284",
            "dark": "#9f6413"
        },
        "tertiary": {
            "light": "#B6E6FC",
            "base": "#065F86",
            "dark": "#185273"
        },
        "text": {
            "lightest": "#FFF",
            "light": "#666",
            "base": "#185273",
            "dark": "#3B3F42",
            "darkest": "#000"
        },
        "background": {
            "lightest": "#FFF",
            "light": "#EFEEEE",
            "base": "#065F86",
            "dark": "#185273",
            "darkest": "#000"
        }
    }
}
