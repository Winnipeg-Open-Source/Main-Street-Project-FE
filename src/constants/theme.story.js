import React from 'react';
import styled from 'styled-components';
import { Flex, Text } from 'pcln-design-system';
import theme from 'constants/theme.json';

const CardContainer = styled(Flex)`
    display: flex;
    flex-direction: column;
    height: 150px;
    box-shadow: 1px 2px 4px 0px rgba(0, 0, 0, 0.25);

    &:hover {
        cursor: pointer;
        box-shadow: 1px 2px 8px 0px rgba(0, 0, 0, 0.5);
    }
`;

function Card ({ title, hexValue }) {
    return (
        <CardContainer width={150} mr={3}>
            <Flex bg={hexValue} style={{ height: 100 }} />
            <Flex flexDirection='column' color='background.lightest' p={2}>
                <Text color='text.darkest'>{title}</Text>
                <Text fontSize={1} color='text.light'>
                    {hexValue}
                </Text>
            </Flex>
        </CardContainer>
    );
}

export const Palette = () => {
    return Object.keys(theme.palette).map((key) => {
        const color = theme.palette[key];
        return (
            <Flex flexDirection='column' p={3}>
                <Text fontSize={3} mb={2}>
                    {key}
                </Text>
                <Flex>
                    {Object.keys(color).map((shade) => {
                        return <Card title={shade} hexValue={color[shade]} />;
                    })}
                </Flex>
            </Flex>
        );
    });
};

export default {
    title: 'Theme',
};
