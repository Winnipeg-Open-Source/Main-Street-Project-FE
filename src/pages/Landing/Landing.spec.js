import React from 'react';
import { fireEvent } from '@testing-library/react';
import renderWithThemeAndRouter from 'tests/helpers/renderWithThemeAndRouter';
import LandingPage from './Landing';
import { LANDING_PATH, DONATIONS_NEW_PATH, REQUISITION_NEW_PATH } from 'constants/paths';

describe('LandingPage', () => {
    it ('renders as expected', () => {
        const { asFragment } = renderWithThemeAndRouter(<LandingPage />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('new donation button works as expected', () => {
        const { getByText, history } = renderWithThemeAndRouter(<LandingPage />, { route: LANDING_PATH });
        const newDonationButton = getByText('New Donation');

        expect(history.location.pathname).toBe(LANDING_PATH);
        fireEvent.click(newDonationButton);
        expect(history.location.pathname).toBe(DONATIONS_NEW_PATH);
    });

    it ('new requisition button works as expected', () => {
        const { getByText, history } = renderWithThemeAndRouter(<LandingPage />, { route: LANDING_PATH });
        const newRequisitionButton = getByText('New Requisition');

        expect(history.location.pathname).toBe(LANDING_PATH);
        fireEvent.click(newRequisitionButton);
        expect(history.location.pathname).toBe(REQUISITION_NEW_PATH);
    });
});
