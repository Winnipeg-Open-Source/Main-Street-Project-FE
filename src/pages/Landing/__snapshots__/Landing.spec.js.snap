// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders as expected 1`] = `
<DocumentFragment>
  <div
    class="sc-qOvoE hIfWnJ sc-qWSGh dpgLb"
  >
    <div
      class="sc-AxirZ sc-AxhCb sc-pIITJ iroxdQ"
      color="primary.light"
      width="1"
    >
      <a
        class="sc-pAMRk fwRIZg"
        href="/donations/new"
      >
        <button
          class="sc-ptEpz iOwfuJ"
          color="primary"
          width="1"
        >
          New Donation
        </button>
      </a>
      <a
        class="sc-pAMRk fwRIZg"
        href="/requisitions/new"
      >
        <button
          class="sc-ptEpz iOwfuJ"
          color="primary"
          width="1"
        >
          New Requisition
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
